com.unity.shadergraph/Editor/Generation/Processors/ShaderStringBuilder.cs

```diff
@@ -203,17 +260,33 @@ public void Concat(ShaderStringBuilder other)
                 currentNode = mapping.node;
 
                 // Use `AppendLines` to indent according to the current indentation.
-                AppendLines(other.ToString(mapping.startIndex, mapping.count));
+                if (m_HumanReadable)
+                {
+                    AppendLines(other.ToString(mapping.startIndex, mapping.count));
```

Comment:
There's also a StringBuilder.Append(StringBuilder, start, count) function that is probably faster than converting to string here.